### PR TITLE
Transcript table

### PR DESCRIPTION
**Pull request by zrose584**
This adds a table which shows the complete transcript, and allows you to seek to each "moment".
So one can ctrl-f audio tracks, which I find very useful. Hope you like it too.

### CODE REVIEW

**Comment by user234683** [comment]
A couple issues:
- For long videos, such as podcasts, the minutes go above 60 and the hour is not displayed
- When captions are off and there's multiple languages ([example video](https://www.youtube.com/watch?v=642x2Y3Zla0)), the language at the top of the list is used. But the user's preferred language is placed at the bottom.
- Same issue as above for the `c` hotkey for turning on captions; it should select the language at the bottom.

**Comment by user234683** [comment]
Also, I suggest having the <details> element containing the transcript be present from the start in the html so it doesn't shift the related videos down when it loads. You can use a jinja if statement to only put it there if there are caption tracks.

**Comment by zrose584** [comment]
>  But the user's preferred language is placed at the bottom.

How do you know that? Just empirical?

**Comment by zrose584** [comment]
> For long videos, such as podcasts, the minutes go above 60 and the hour is not displayed

yt.com itself has the same "bug"/problem (in their transcript table), so I don't think it's worth fixing..

**Comment by user234683** [comment]
No need to duplicate Youtube's bugs. I suggest fixing it. It should be a quick fix to the toMS function. If you want, you could only include the hour digits if the video is an hour or longer if you think having 0: before everything for short videos would be ugly.

My other two suggestions before merging are to (1) include a label next to the checkbox so its clear that it toggles contiguous view and (2) include `background-color: var(--interface-color)` and a little bit of padding in the css for the div that holds the transcript cues (you can see why this is important by switching to dark or gray theme).

An additional problem I found is that the code isn't detecting automatic captions format on Firefox because there's a bug (?) in Firefox's VTT parsing where it will ignore newlines. So if the cue starts with a newline, that cue will have blank text (a corollary is that the first sentence uttered will fail to display in the automatic captions in Firefox until the next cue). So automatic captions include all the `<c>` vomit in the transcript on Firefox. So using a different method to detect automatic captions should be a quick fix (though the first line of the transcript will be missing). Alternatively, there's already a route in youtube-local that generates a simple transcript text file based on the captions. It's located in the downloads dropdown. This transcript file gets rid of the `<c>` stuff and other things. 

For future pull requests, coding-style wise, I suggest increasing the verbosity and descriptivity of the variable names and not using single-letter variable names (with the exception of iterator variables such as i,j and the like). For instance,
- toMS -> toMinSec (or toHourMinSec or toTimestamp)
- createA -> createTimestampAnchor or createTimestampLink
- tt -> textTracks or captionTracks (I kept forgetting what tt and tts meant in each context and it got hard to follow)
- tt_type -> is_automatic or is_automatic_captions
- cuesL -> cuesList
- cb -> I'm unsure what this stands for
- idxC -> cuesIndex (I haven't seen idx before but I'm willing to be lenient if it's a common abbreviation)

etc.

And I suggest using comments in areas where it's not clear to the unacquainted why something is being done. For instance, I would recommend putting a comment above `if (i % 2) continue;` briefly explaining that this is skipping duplicate lines in automatic caption format.

**Comment by zrose584** [comment]
> Firefox's VTT parsing is buggy

What a pain.. Can you send a bug report to them? I believe you know the "domain specific" terms better than me.
